.github/workflows/duplicate-external-pr.yml

```diff
@@ -54,14 +55,23 @@ jobs:
               if: steps.check-branch.outputs.branch_exists == 'false'
               env:
                   ISSUE_NUMBER: ${{ github.event.issue.number }}
-                  PR_TITLE: ${{ github.event.issue.title }}
-                  REF: ${{ github.event.issue.pull_request.head.ref }}
+                  PR_REPO: ${{ github.event.issue.pull_request.head.repo.full_name }}
+                  PR_BRANCH: ${{ github.event.issue.pull_request.head.ref }}
                   DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
                   ASSIGNEE: ${{ github.event.comment.user.login }}
                   GITHUB_TOKEN: ${{ secrets.CI_GITHUB_TOKEN }}
               run: |
                   git config user.name "github-actions[bot]"
                   git config user.email "github-actions[bot]@users.noreply.github.com"
-                  git checkout -b duplicate-${{env.ISSUE_NUMBER}} ${{env.REF}}
-                  git push origin duplicate-${{env.ISSUE_NUMBER}}
-                  gh pr create --base ${{env.DEFAULT_BRANCH}} --head duplicate-${{env.ISSUE_NUMBER}} --title "chore: ${{env.ISSUE_NUMBER}} [DUPLICATE]" --body "This is a duplicated PR for running E2E tests." --assignee ${{env.ASSIGNEE}}
+                  echo "PR_REPO ${{env.PR_REPO}}"
+                  echo "PR_BRANCH ${{env.PR_BRANCH}}"
+                  echo "DEFAULT_BRANCH ${{env.DEFAULT_BRANCH}}"
+                  echo "Adding remote for external PR repo"
+                  git remote add external https://github.com/${{env.PR_REPO}}.git
+                  echo "Fetching from external PR repo"
+                  git fetch external
+                  echo "Checking out and creating duplicate branch from PR branch"
+                  git checkout -b duplicate_${{env.PR_BRANCH}} external/${{env.PR_BRANCH}}
```

Comment:
I think we can reference `external` here instead of using `refs/remotes/...` (like we have in our internal script